[DOCS] Support for ECS 1.11 indicator data (#1444)
elastic/security-docs: docs/detections/rules-ui-create.asciidoc

Additionally, all future exceptions added to the Elastic Endpoint Security rule
also affect this rule.
+

.. *Building block* (optional): Select to create a building-block rule. By
default, alerts generated from a building-block rule are not displayed in the
UI. See <<building-block-rule>> for more information.

> .. *Indicator prefix override*: Define the data structure path of indicator data inside indicator documents. When the indicator match rule executes, it queries specified indicator indices and references this setting to locate fields with indicator data. This data is used to enrich indicator match alerts with metadata about matched threat indicators. The default value for this field is `threat.indicator`.

Reviewer: maybe we can say something like `Define the location of enrichment data within the structure of the indicator document`. Because there is all kinds of data in the indicators that are outside this path as well, but only what's within the threat_indicator_path will be used as the enrichment

.. *Building block* (optional): Select to create a building-block rule. By
default, alerts generated from a building-block rule are not displayed in the
UI. See <<building-block-rule>> for more information.

> .. *Indicator prefix override*: Specify the default threat indicator path that the indicator match rule will use when obtaining threat indicator data from the defined indicator indices. This data is used to enrich indicator match alerts with metadata about matched threat indicators. The default value for this field is `threat.indicator`. 

Reviewer 2: This is more a question than a suggestion, but I would have thought it would be "threat indicator index" not "threat indicator path", but my searches didn't turn up many references to either one. Why is it threat indicator path instead of index?

Author: @benironside, my understanding is that this setting tells the rule the _path location_ of threat indicator fields. So, when the rule queries the indicator indices--which are specified when you create the rule (see step 1c [here](https://www.elastic.co/guide/en/security/master/rules-ui-create.html#create-indicator-rule))-- it knows how to identify fields with indicator data. @ecezalp please correct me if I'm wrong about any of this or missing anything!

Would it be more clear if I added `location` after `default threat indicator path` and changed `obtaining` to `searches for`?

Reviewer: the value of `indicator_index` is used to determine which indices will be queried to find the indicator documents.
the value of `threat_indicator_path` is used to point out where the enrichment data is in a single indicator document.

so in ECS 1.11 a single incidator event roughly looks like what's below.
```
{
  agent: {name: "ece's computer"},
  event: {ingested: "this-date", category: "indicator"},
  threat: {indicator: {url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}}}},
  @timestamp: "timestampvalue"
}
```
then we have an event we want to enricht that look like 
```
{
  threat: enrichments: []
}
```
we find `threat_indicator_path` on the indicator, which is `threat.indicator`, which gives us the enrichment: `{url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}` there.

once the event is enriched with that  it looks like this
```
{
  threat: enrichments: [{matched: {id:"asdf", value: "asdf", type: "asdf"}, indicator: {ENRICHMENT_GOES_HERE }}]
}
```

so an enriched event will have `{url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}}` placed into `threat.enrichments[0].indicator`. if event already had another enrichment then it'll go to `threat.enrichments[1].indicator`

before indicator data had different mappings and it looked like the following, so we used `threatintel.indicator` as the `threat_indicator_path`
```
{
  agent: {name: "ece's computer"},
  event: {ingested: "this-date", category: "indicator"},
  threatintel: {indicator: {url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}}}},
  @timestamp: "timestampvalue"
}
```

for something custom that users may want to use threats could look like anything
```
{
  random: {extrarandom: {url: {full: "www.asdf.com"}, file: {hash: {md5: "asdfasd", sha1: "asdfasgsdfg"}}}},
  @timestamp: "timestampvalue"
}
```
than the threat_indicator_path should configured to be `random.extrarandom`. if user uses non ECS fields and try to read `threat.indicator` as the threat_indicator_path it might just resolve to `undefined` so there will be no data enriched on the event.

Reviewer 3: I think I had a similar reaction to @benironside -- when I see "path" I usually think of a file path, or a URL path. Thank you @ecezalp for the explanation! Now I understand (i think?) we're talking about a path through the data structure within the indicator event's data. How about something like:
```suggestion
.. *Indicator prefix override*: Specify the default threat indicator data structure path that the indicator match rule will use to locate threat indicator data in the defined indicator indices. This data is used to enrich indicator match alerts with metadata about matched threat indicators. The default value for this field is `threat.indicator`. 
```

Author: I like the suggestion to specify that we're talking about the data structure--"path" is pretty vague and can be even more so for newer users. Thanks all for the input!

+

> IMPORTANT: If your threat indicator data is at a different location, update this setting with the appropriate field prefix to ensure alert enrichment can still be performed.

Reviewer 2: ```suggestion
IMPORTANT: If your threat indicator data is at a different location, update this setting with the appropriate field prefix to enable alert enrichment.
```

Author: I like the succinctness of your recommendation, but am concerned it might give the impression that this setting can be turned on and off, and we really don't ever want users to turn it off or leave the setting undefined. I believe the main idea we're trying to convey here is that, if users have non-ECS-compliant indices, their threat fields will be located at a different path. So, to ensure they're still able to view threat intel on their indicator match alerts, they should define that custom path here.  @ecezalp feel free to correct me if I'm wrong.

Reviewer: +1 to `ensure alert enrichment can still be performed` because threat_indicator_path is a mandatory field

.. *Building block* (optional): Select to create a building-block rule. By
default, alerts generated from a building-block rule are not displayed in the
UI. See <<building-block-rule>> for more information.

> .. *Indicator prefix override*: Specify the default threat indicator path (`threat_indicator_path`) that the indicator match rule will use when obtaining threat indicator data from the default threat intelligence index. This data is used to enrich indicator match alerts with metadata about matched threat indicators. The default value for this field is `threat.indicator`. If your threat indicator data is at a different location, update this setting to ensure the correct field prefix is used so alert enrichment can still be performed.

Reviewer: first sentence - indicator match rule will use threat_indicator_path to obtain indicator data NOT from the default threat index, but from the indicators obtained from the value of `indicator index` specified in the rule, could be default or custom.

Author: Ah yes, the IM rule queries threat indicator indices defined on the **Indicator index patterns** field

+

> NOTE: When upgrading to {stack} version 8.x, the default threat indicator path of an indicator match rule created in {stack} 7.x is automatically updated to `threatintel.indicator`. This allows the rule to continue using indicator data ingested by {filebeat} version 7.x. To make a rule use threat indicator data ingested from {agent} or {filebeat} version 8.x, enter the `threat.indicator` value into the *Indicator prefix override* field. Separate rules must be created to query threat intelligence indices created by {filebeat} version 7.x and version 8.x because different default threat indicator path values are required.

Reviewer: first sentence: the value is automatically updated to `threatintel.indicator` IF it wasn't previously specified. if the users had provided a custom value that won't be overwritten.

Reviewer: third sentence: users won't have to manually enter `threat.indicator` if they are creating a new rule, we auto-populate that by default in 8.x. If they had created a rule in 7.x with filebeat, and they upgrade both kibana and filebeat to 8.0, only then they would have to come back to their old rule and update `threatintel.indicator` to `threat.indicator`
